### DISCUSSION TITLE
How to make draggable table column headers?

### DISCUSSION THREAD

**Discussion by jarruda** [emilk/egui#4737] [Q&A]
I would like to make the entire column header cell a drag and drop item so I can implement column re-ordering.

If I attempt to use allocate_rect(min_rect), or I use a right-to-left layout within the cell, the cell just grows infinitely.

How can I have the whole cell be a draggable area? Aside: is it possible to align some things (e.g. a button) to the right without growing the column?

![image](https://github.com/emilk/egui/assets/714104/b0400604-f779-4d7b-899d-76119835800c)

**Comment by YgorSouza** [comment]
I've never done that, but I know this project implements column reordering: <https://github.com/kang-sw/egui-data-table>. It uses a custom type rather than the Table from egui_extras, but maybe you can look at their implementation for inspiration. The drag-and-drop part should be around [here](https://github.com/kang-sw/egui-data-table/blob/f4283b163343f2481840658aa5b41fc92ae34d69/src/draw.rs#L144).
